src/storage/secondary/merge_iterator.rs

```diff
@@ -48,19 +50,25 @@ impl MergeIterator {
         (left_id, left_batch_row_id): &(usize, usize),
         (right_id, right_batch_row_id): &(usize, usize),
     ) -> std::cmp::Ordering {
-        let left_data = self.chunk_buffer[*left_id]
-            .as_ref()
-            .unwrap()
-            .array_at(self.sort_key_idx)
-            .get(*left_batch_row_id);
-        let right_data = self.chunk_buffer[*right_id]
-            .as_ref()
-            .unwrap()
-            .array_at(self.sort_key_idx)
-            .get(*right_batch_row_id);
-
-        // TODO: handle can-not-compare
-        left_data.partial_cmp(&right_data).unwrap()
+        let mut res = Ordering::Equal;
+        for sort_key_idx in &self.sort_key_idx {
+            let left_data = self.chunk_buffer[*left_id]
+                .as_ref()
+                .unwrap()
+                .array_at(*sort_key_idx)
+                .get(*left_batch_row_id);
+            let right_data = self.chunk_buffer[*right_id]
+                .as_ref()
+                .unwrap()
+                .array_at(*sort_key_idx)
+                .get(*right_batch_row_id);
+            // TODO: handle can-not-compare
+            res = left_data.partial_cmp(&right_data).unwrap();
+            if Ordering::Equal != res {
+                break;
```

Comment:
just return res here? so that `res` can simply be a local variable

Reply:
Ok